python/SWESimulators/KP07.py

```diff
@@ -215,117 +257,130 @@ def cleanUp(self):
-                        self.wind_stress.u0, self.wind_stress.v0, \
-                        self.boundary_conditions.north, self.boundary_conditions.east, self.boundary_conditions.south, self.boundary_conditions.west, \
-                        self.t)
-                
-                self.bc_kernel.boundaryCondition(self.cl_queue, \
-                        self.cl_data.h0, self.cl_data.hu0, self.cl_data.hv0) 
-            else:
-                self.kp07_kernel.swe_2D(self.cl_queue, self.global_size, self.local_size, \
-                        self.nx, self.ny, \
-                        self.dx, self.dy, local_dt, \
-                        self.g, \
-                        self.theta, \
-                        self.f, \
-                        self.coriolis_beta, \
-                        self.y_zero_reference_cell, \
                         self.r, \
-                        np.int32(0), \
-                        self.cl_data.h0.data,  self.cl_data.h0.pitch,  \
-                        self.cl_data.hu0.data, self.cl_data.hu0.pitch, \
-                        self.cl_data.hv0.data, self.cl_data.hv0.pitch, \
-                        self.cl_data.h1.data,  self.cl_data.h1.pitch,  \
-                        self.cl_data.hu1.data, self.cl_data.hu1.pitch, \
-                        self.cl_data.hv1.data, self.cl_data.hv1.pitch, \
-                        self.bathymetry.Bi.data, self.bathymetry.Bi.pitch, \
-                        self.bathymetry.Bm.data, self.bathymetry.Bm.pitch, \
-                        self.wind_stress.type, \
-                        self.wind_stress.tau0, self.wind_stress.rho, self.wind_stress.alpha, self.wind_stress.xm, self.wind_stress.Rc, \
-                        self.wind_stress.x0, self.wind_stress.y0, \
-                        self.wind_stress.u0, self.wind_stress.v0, \
-                        self.boundary_conditions.north, self.boundary_conditions.east, self.boundary_conditions.south, self.boundary_conditions.west, \
-                        self.t)
+                        self.var_dt.data, \
+                        np.int32(substep) )
+                        
+    def boundaryConditionsKernel(self, U1, U2, U3):
+        self.bc_kernel.boundaryCondition(self.cl_queue, U1, U2, U3)
+        
+    def step(self, t_end=0.0):
+        """
+        Function which steps n timesteps
+        """
+
+        # Only update dt if we are using variable/computed dt
+        update_dt = 1
+        if self.dt is not None:
+            update_dt = 0
+            local_dt = np.float32(self.dt)
+
+        n = 0
+        t_end = np.float32(t_end);
+        while (t_end > 0.0):
+            if (self.use_rk2):
+                # Substep one
+                self.fluxKernel(update_dt, self.cl_data.h0, self.cl_data.hu0, self.cl_data.hv0)
+                if self.dt is not None:
+                    local_dt = np.float32(min(t_end, local_dt))
+                    upload_dt = np.zeros((128), dtype=np.float32)
```

Comment:
Why hardcoded 128 here? Should this be dt_block_size or similar?

Reply:
I think this should be just 1, and was set to 128 for testing.

Reply:
Probably. Uncritical copy-pasting from my part.

Reply:
And uncritical commit from me. No biggie, but should be fixed in the future. Will crash for small domains, I think (if the device size buffer is less than 128 long)